src/prog_models/sim_result.py

```diff
@@ -92,18 +111,18 @@ def remove(self, d : dict = None, t : float = None) -> None:
             raise ValueError("ValueError: Only one named argument (d, t) can be specified.")
 
         if (t is not None):
-            self.data.pop(self.times.index(t))
-            self.times.remove(t)
+            target_index = self.times.index(t)
         else:
-            self.times.pop(self.data.index(d))
-            self.data.remove(d)
+            target_index = self.data.index(d)
+        self.pop(target_index)
 
     def clear(self) -> None:
         """Clear the SimResult"""
         self.times = []
         self.data = []
+        self._frame = None
 
-    def time(self, index : int) -> float:
+    def get_time(self, index: int) -> float:
```

Comment:
What's the reason for this change? This will change the interface.

Reply:
It was so confusing when looking at self.time(x) and self.time[x] side by side that I felt like that change would help with readability. It's also following the python naming convention

Reply:
I disagree with your statement about this following python naming conventions though. I think this is more java naming conventions, python prefers to use the name rather than get_name. 

From https://python-consistency.readthedocs.io/en/latest/conventions.html:
"* Do not use get_... Prefer to use the name without the get_ prefix."

That said, I see what you mean about the confusion of self.times[x] and self.time(x). And with times being accessible, this function is also redundant. Why don't we instead move towards making users access times[x] only? We can keep time for one release (per ProgPy policy) with a deprecation warning telling users to use .times[x] instead. What do you think?